src/PJ_helmert.c

```diff
@@ -245,196 +468,102 @@ PJ *PROJECTION(helmert) {
-        should always check which convention (exact or approximate rotation
-        matrix) is expected, and whether the induced error for selecting
-        the opposite convention is acceptable (which it often is).
 
+    /* Scale rate */
+    if (pj_param (P->ctx, P->params, "tds").i)
+        Q->dscale = pj_param (P->ctx, P->params, "dds").f;
 
-        Sign conventions
-        ----------------
 
-        Take care: Two different sign conventions exist.
+    /* Epoch */
+    if (pj_param(P->ctx, P->params, "tepoch").i)
+        Q->epoch = pj_param (P->ctx, P->params, "depoch").f;
 
-        Conceptually they relate to whether we rotate the coordinate system
-        or the "position vector" (the vector going from the coordinate system
-        origin to the point being transformed, i.e. the point coordinates
-        interpreted as vector coordinates).
+    if (pj_param(P->ctx, P->params, "ttobs").i)
+        Q->t_obs = pj_param (P->ctx, P->params, "dtobs").f;
 
-        Switching between the "position vector" and "coordinate system"
-        conventions is simply a matter of switching the sign of the rotation
-        angles, which algebraically also translates into a transposition of
-        the rotation matrix.
-
-        Hence, as geodetic constants should preferably be referred to exactly
-        as published, the "transpose" option provides the ability to switch
-        between the conventions.
+    /* Use small angle approximations? */
+    if (pj_param (P->ctx, P->params, "bapprox").i)
+        Q->approximate = 1;
 
-    ***************************************************************************/
-    {
-        double  f,  t,  p;    /* phi/fi , theta, psi  */
-        double cf, ct, cp;    /* cos (fi, theta, psi) */
-        double sf, st, sp;    /* sin (fi, theta, psi) */
-
-        /* rename   (omega, phi, kappa)   to   (fi, theta, psi)   */
-        f = opk.o;
-        t = opk.p;
-        p = opk.k;
-
-        if (approximate) {
-            R00 =  1;
-            R01 =  p;
-            R02 = -t;
-
-            R10 = -p;
-            R11 =  1;
-            R12 =  f;
-
-            R20 =  t;
-            R21 = -f;
-            R22 =  1;
-        }
-        else {
-            cf = cos(f);
-            sf = sin(f);
-            ct = cos(t);
-            st = sin(t);
-            cp = cos(p);
-            sp = sin(p);
-
-
-            R00 = ct*cp;
-            R01 = cf*sp + sf*st*cp;
-            R02 = sf*sp - cf*st*cp;
-
-            R10 = -ct*sp;
-            R11 =  cf*cp - sf*st*sp;
-            R12 =  sf*cp + cf*st*sp;
-
-            R20 =  st;
-            R21 = -sf*ct;
-            R22 =  cf*ct;
-        }
+    /* Use "other" rotation sign convention? */
+    if (pj_param (P->ctx, P->params, "ttranspose").i)
+        Q->transpose = 1;
+
+    Q->xyz    =  Q->xyz_0;
+    Q->opk    =  Q->opk_0;
+    Q->scale  =  Q->scale_0;
+
+    /* Let's help with debugging */
+    pj_log_debug(P, "Helmert parameters:");
```

Comment:
Same comment as above, regarding enclosing in if (PJ_LOG...) { ...}.

I agree that this logging info is very important when implementing new transformations, so the comment about helping with debugging is highly appropriate